syntax = "proto3";
import "google/protobuf/empty.proto";


service UserService {
  rpc GetUserByID (GetUserByIDRequest) returns (BaseUserDataResponse);
  rpc GetAllUsers (GetAllUsersRequest) returns (GetAllUsersResponse);
  rpc CreateUser (CreateUserRequest) returns (google.protobuf.Empty);
  rpc UpdateByIdUser (UpdateUserByIDRequest) returns (google.protobuf.Empty);
  rpc DeleteUser (DeleteUserByIDRequest) returns (google.protobuf.Empty);
}

message GetUserByIDRequest {
  string user_id = 1;
}

message BaseUserDataResponse {
  string user_id = 1;
  string email = 2;
  string first_name = 3;
  string last_name = 4;
}

message GetAllUsersRequest {

}

message GetAllUsersResponse {
  repeated BaseUserDataResponse users_data = 1;
}

message CreateUserRequest {
  string email = 1;
  string first_name = 2;
  string last_name = 3;
  string phone_number = 4;
  string password = 5;
}

message UpdateUserByIDRequest {
  string user_id = 1;
  string first_name = 2;
  string last_name = 3;
  string phone_number = 4;
  string password = 5;
}

message DeleteUserByIDRequest {
    string user_id = 1;
}